pkg/node/manager/manager.go

```diff
@@ -167,6 +180,24 @@ func (m *manager) AddNode(nodeName string) error {
 	return nil
 }
 
+func (m *manager) CreateCNINodeIfNotExisting(node *v1.Node) error {
+	if cniNode, err := m.wrapper.K8sAPI.GetCNINode(node.Name, config.KubeDefaultNamespace); err != nil {
+		m.Log.Info("Failed finding CNINode as a creating check", "Error", err)
+		if apierrors.IsNotFound(err) {
+			m.Log.Info("Will create a new CNINode", "CNINodeName", node.Name)
+			return m.wrapper.K8sAPI.CreateCNINode(node)
+		}
+		return err
+	} else {
+		if cniNode != nil && cniNode.Name == node.Name {
+			m.Log.Info("The CNINode is already existing", "CNINode", cniNode)
+		} else {
+			m.Log.Error(errors.New("API error caused unmatched CNINode"), "Found existing unmatched CNINode when creating CNINode", "NodeName", node.Name)
```

Comment:
How could the name not match?

Reply:
Some race condition, users' mis-operation, or other causes?

Reply:
On line 184: https://github.com/aws/amazon-vpc-resource-controller-k8s/pull/270/files#diff-e9cec4c2460e089f0f518c07f3d4c2fb40273486a8346d499ec5188395417772R184, we retrieve the node by name. The only way that the name wouldn't match, is if our client is fundamentally broken, and we have bigger problems.

Reply:
Agree. I will remove the redundant check.